File: backend/app/models/investigation.py
from typing import Any

from pydantic import BaseModel


class InvestigationResponse(BaseModel):
    status: str
    diagnosis: dict[str, Any]
    investigation: dict[str, Any]


class InvestigationRequest(BaseModel):
    kube_context: str | None = None
    realtime_channel: str | None = None


class ClusterContextsResponse(BaseModel):
    status: str
    contexts: list[str]
    current_context: str = ""
    error: str = ""

File: frontend/types/investigation.ts
export type Diagnosis = {
  root_cause: string;
  explanation: string;
  fix: string;
  kubectl_commands: string[];
  kubectl_command: string;
  prevention: string;
  confidence: number;
  confidence_reasoning: string;
};

export type InvestigationResponse = {
  status: string;
  diagnosis: Diagnosis;
  investigation: Record<string, unknown>;
};

export type ClusterContextsResponse = {
  status: string;
  contexts: string[];
  current_context: string;
  error: string;
};

export type InvestigationHistoryItem = {
  id: string;
  root_cause: string;
  namespace: string | null;
  confidence: number;
  status: string;
  created_at: string;
};
